src/black/files.py

```diff
@@ -118,8 +119,13 @@ def get_gitignore(root: Path) -> PathSpec:
     gitignore = root / ".gitignore"
     lines: List[str] = []
     if gitignore.is_file():
-        with gitignore.open() as gf:
+        with open(gitignore, "rb") as buf:
+            srcbuf = io.BytesIO(buf.read())
+            encoding, _ = tokenize.detect_encoding(srcbuf.readline)
```

Comment:
`tokenize.detect_encoding` detects the encoding based on the approach used in Python files (`# encoding: utf-8` comments), so I don't think this works for gitignore files.

Reply:
And the reason why this code works also well in @temeddix's case is that `tokenize.detect_encoding` returns `utf-8` if no encoding indicator was found. So in a way this is an overengineered way of saying `open(gitignore, "r", encoding="utf-8")`. https://docs.python.org/3/library/tokenize.html#tokenize.detect_encoding